chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/ProcessPeaksByPeakClassifierFilter.java

```diff
@@ -0,0 +1,169 @@
+		LinkedHashMap<String, String> parsedUserDefinedValues = parseUserDefinedValuesAsMap(configuration);
+
+		for(X peak : read) {
+			setPeakClassifier(configuration, parsedUserDefinedValues, peak);
+			subMonitor.worked(1);
+		}
+	}
+
+	private static LinkedHashMap<String, String> parseUserDefinedValuesAsMap(ProcessPeaksByPeakClassifierFilterSettings configuration) {
+
+		String userDefinedMatchExpression = configuration.getUserDefinedMatchExpression();
+		String matchClassification = configuration.getMatchClassification();
+		List<List<String>> matchList = new ArrayList<List<String>>();
+		if(!userDefinedMatchExpression.isEmpty()) {
+			parseExpressions(matchList, userDefinedMatchExpression);
+		} else {
+			throw new IllegalArgumentException("No match expressions defined!");
+		}
+		if(!matchClassification.isEmpty()) {
+			parseExpressions(matchList, matchClassification);
+		} else {
+			throw new IllegalArgumentException("No match classifications defined!");
+		}
+		if(matchList.get(0).size() != matchList.get(1).size()) {
+			throw new IllegalArgumentException("The number of Expressions is not the same as the number of Classifications.");
+		}
+		LinkedHashMap<String, String> parsedUserDefinedValues = new LinkedHashMap<String, String>();
+		for(int i = 0; i < matchList.get(0).size(); i++) {
+			parsedUserDefinedValues.put(matchList.get(0).get(i),matchList.get(1).get(i));
+		}
+		return parsedUserDefinedValues;
+	}
+
+	private static void parseExpressions(List<List<String>> matchList, String userExpressions) {
+
+		List<String> expressionsList = new ArrayList<String>();
+		String[] tempExpressions = userExpressions.split(System.getProperty("line.separator"));
+		for(int i = 0; i < tempExpressions.length; i++) {
+			if(!tempExpressions[i].isEmpty()) {
+				expressionsList.add(tempExpressions[i]);
+			}
+		}
+		matchList.add(expressionsList);
+	}
+
+	private static <X extends IPeak> void setPeakClassifier(ProcessPeaksByPeakClassifierFilterSettings configuration, LinkedHashMap<String, String> parsedUserDefinedValues, X peak) {
+
+		peak.removeClassifier("");
+		String substanceName = extractSubstanceName(configuration, peak);
+		for(Map.Entry<String, String> entry : parsedUserDefinedValues.entrySet()) {
+			Pattern pattern = createPattern(configuration, entry);
+			Matcher matcher = pattern.matcher(substanceName);
+			/*
+			 * UserDefinedMatchClassification for the cases when
+			 * 1) the occurrence of the regular expression is found in identified name or
+			 * 2) the whole text is found in identified name
+			 */
+			if(configuration.isWildcardSearch()) {
+				if(matcher.find()) {
+					peak.addClassifier(entry.getValue());
+				}
+			} else {
+				if(matcher.matches()) {
+					peak.addClassifier(entry.getValue());
+				}
+			}
+		}
+	}
+
+	private static Pattern createPattern(ProcessPeaksByPeakClassifierFilterSettings configuration, Map.Entry<String, String> entry) {
+
+		Pattern pattern = null;
```

Comment:
You could also create the Pattern case-insensitive.
Pattern.compile(regexp, Pattern.CASE_INSENSITIVE)

Reply:
There is always more than one possibility to achieve the goal. I will incorporate this possibility.